ui/pages/token-allowance/token-allowance.js

```diff
@@ -122,24 +125,25 @@ export default function TokenAllowance({
   const unapprovedTxCount = useSelector(getUnapprovedTxCount);
   const unapprovedTxs = useSelector(getUnapprovedTransactions);
   const useCurrencyRateCheck = useSelector(getUseCurrencyRateCheck);
-  let customTokenAmount = useSelector(getCustomTokenAmount);
-  if (thisOriginIsAllowedToSkipFirstPage && dappProposedTokenAmount) {
-    customTokenAmount = dappProposedTokenAmount;
+  const customTokenAmount = useSelector(getCustomTokenAmount);
+  if (!thisOriginIsAllowedToSkipFirstPage && !dappProposedTokenAmount) {
```

Comment:
this conditional is not the exact opposite of the former. was this intentional?

Reply:
Thanks for bringing this up ❤️ With the current changes, we default to dappProposedTokenAmount when we load the token allowance page. That is the reason I changed the previous conditional check. But your comment made me think, and I need to do more testing to decide whether the current condition I added is relevant.